std/experimental/checkedint.d

```diff
@@ -472,6 +472,48 @@ if (isIntegral!T || is(T == Checked!(U, H), U, H))
         assert(!MyHook.thereWereErrors);
     }
 
+    // toHash
+    /**
+
+    Generates a hash for `this`. If `Hook` defines `hookToHash`, the call
+    immediately returns `hook.hookToHash(payload)`. If `Hook` does not
+    implement `hookToHash`, but it has state, a hash will be generated for
+    the `Hook` using the built-in function and it will be xored with the
+    hash of the `payload`. Otherwise, the hash of the payload is returned.
+
+    */
+    size_t toHash() const nothrow @safe
```

Comment:
Please remove the function attributes as they should be inferred from `Hook` and/or `payload`

Reply:
`toHash` is used by the associative array and the hashing function has to be `const nothrow @safe`. This will not work with `aa`, unless we force the `Hook` to respect the same attributes.
Instead of letting `toHash` infer attributes from the `Hook`, I think it's better for `toHash` to force the `Hook` to be `const nothrow @safe`.

Reply:
Cool. Thank you for the insight. I hadn't look at `AA`. Let's leave this as it is then.

I think we should still add a ddoc unittest that builds an AA of Checked